src/Storages/ObjectStorage/DataLakes/DeltaLake/KernelUtils.cpp

```diff
@@ -30,31 +36,93 @@ void * KernelUtils::allocateString(ffi::KernelStringSlice slice)
 
 namespace
 {
+
 struct KernelError : public ffi::EngineError
 {
     std::string error_message;
 };
+
+std::mutex mutex;
+std::unordered_set<uintptr_t> allocated_errors;
```

Comment:
You followed my Idea, that is great and save, good.

But with this approach we do no need wrong inheritance any more, just store the attached data inside external container.

Reply:
removed inheritance